components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -386,6 +403,77 @@ void AdsServiceImpl::Start() {
   BackgroundHelper::GetInstance()->AddObserver(this);
 }
 
+void AdsServiceImpl::MaybeShowFirstLaunchNotification() {
+  if (!ShouldShowAdsNotification() ||
+      !profile_->GetPrefs()->GetBoolean(
+          brave_rewards::prefs::kBraveRewardsEnabled))
+    return;
+
+  if (profile_->GetPrefs()->GetTime(
+          prefs::kBraveAdsLaunchNotificationTimestamp) == base::Time()) {
+      profile_->GetPrefs()->SetTime(
+          prefs::kBraveAdsLaunchNotificationTimestamp,
+        base::Time::Now());
+  }
+
+  auto* rewards_service =
+      brave_rewards::RewardsServiceFactory::GetForProfile(profile_);
+
+  auto* rewards_notification_service =
+      rewards_service->GetNotificationService();
+
+  RewardsNotificationService::RewardsNotificationArgs args;
+  rewards_notification_service->AddNotification(
+      RewardsNotificationService::REWARDS_NOTIFICATION_ADS_LAUNCH,
+      args,
+      rewards_notification_ads_launch);
+
+  uint32_t timeout_length = IsTestingEnv()
+      ? (5 * 60) /* 5 minutes */
+      : (24 * 60 * 60 * 7); /* 7 days */
+
+  auto timeout = base::Time::Now() - profile_->GetPrefs()->GetTime(
```

Comment:
Shouldn't this calculation be `(kBraveAdsLaunchNotificationTimestamp + timeout) - Now()` and if < 0 set to 0 so that the timer is fired instantly

Reply:
resolved (nofix)

Reply:
updated to new calc